In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

### Simulating a real world phenomena

Chosen phenom - physical health of the individual, based off results from a pseudo-survey.

In essence I hope to simulate the results of a survey asking $n$ number of people about their health and lifestyle and attributing a grade of their general health. From the created dataset we would hope to predict the overall health of the inividual form their own inputs to the survey.

Age

Sex

Height/Weight

Drink(units/week)

Smoker (cigarettes/day)

Previous smoker( length of time somking)

Hours of exercise(/week)

Consistency of exercise (regular exercise)

Diet( categorise)

Calories eaten per day

Average hours of sleep

Water drank/ day

Underlying health condition


So our first item it tocreate a dateset with these criteria as column headers.

In [203]:
columns=['Age', 'Sex','Height', 'Weight', 'Drinker', 'Alcohol_units_wk','Smoker', 'Cigs/day', 'Time_Smoked_Prev' 'Exercise_wk', 'Exercise_Consistency', 'Diet',
                           'Calories_day','Hours_Sleep', 'Water_l_day', 'Stress_lvls', 'Underlying_health']

In [3]:
df = pd.DataFrame(columns = columns)

In [4]:
df

Empty DataFrame
Columns: [Age, Sex, Height, Weight, Alcohol_units_wk, Smoker, Cigs/day, Time_Smoked_PrevExercise_wk, Exercise_Consistency, Diet, Calories_day, Hours_Sleep, Water_l_day, Stress_lvls, Underlying_health]
Index: []

In [5]:
rng = np.random.default_rng()

In [9]:
age = rng.integers(18, 55, size = 100)
age2  = rng.choice((age),100)

In [12]:
age

array([32, 18, 31, 51, 54, 37, 37, 34, 19, 21, 38, 31, 26, 48, 37, 39, 37,
       18, 42, 54, 19, 32, 42, 44, 53, 49, 35, 44, 35, 47, 46, 26, 40, 33,
       24, 20, 53, 30, 34, 38, 25, 50, 31, 43, 42, 21, 23, 32, 23, 32, 47,
       23, 25, 34, 54, 33, 26, 37, 20, 42, 47, 23, 35, 31, 48, 49, 52, 43,
       33, 28, 45, 25, 43, 51, 50, 39, 45, 54, 45, 20, 52, 33, 22, 28, 33,
       48, 21, 36, 51, 45, 28, 37, 35, 43, 53, 38, 26, 43, 28, 23],
      dtype=int64)

In [156]:
def age():
    return rng.integers(18, 60)    

In [157]:
age()

52

In [10]:
age2

array([37, 43, 33, 54, 37, 25, 53, 26, 37, 52, 47, 21, 26, 21, 31, 42, 21,
       42, 37, 28, 43, 37, 52, 43, 37, 30, 42, 54, 39, 36, 30, 20, 37, 20,
       48, 50, 42, 34, 53, 37, 44, 31, 32, 45, 32, 26, 37, 35, 25, 30, 35,
       54, 45, 20, 32, 35, 45, 53, 53, 54, 47, 34, 37, 19, 33, 51, 25, 45,
       37, 20, 32, 26, 20, 51, 38, 53, 21, 31, 48, 52, 53, 33, 53, 47, 51,
       33, 20, 47, 43, 52, 53, 18, 53, 48, 26, 53, 42, 50, 32, 23],
      dtype=int64)

In [14]:
sexes = ['Male', 'Female']
sex  = rng.choice((sexes),100)

In [115]:
def sexes():
    sexes = ['Male', 'Female']
    sex  = rng.choice((sexes))
    return sex

In [120]:
sexes()

'Male'

In [19]:
height_female = rng.normal(165, 7, 100)
height_female

array([168.19984627, 168.34303692, 167.44743386, 163.23487937,
       164.45153955, 162.54580692, 164.94760097, 157.73643257,
       154.67646746, 160.92364328, 168.26227415, 152.71468289,
       164.98504291, 171.17748172, 172.50481376, 161.45212426,
       167.92113569, 166.44915643, 159.51154875, 172.23256371,
       162.46625513, 169.05856329, 158.17368888, 147.64904878,
       161.89360177, 170.56093527, 160.704677  , 162.66046272,
       165.35910398, 180.80856471, 167.89074573, 164.07488173,
       153.16832982, 164.26625204, 159.4579111 , 174.36520815,
       151.96964574, 155.23523399, 177.38367286, 172.77028448,
       167.25532678, 172.00323758, 163.13998578, 170.5967611 ,
       152.35333523, 181.67811506, 163.49811403, 158.95800108,
       160.90760384, 168.88765276, 157.76407525, 154.69780358,
       161.68389243, 172.77956764, 171.74512576, 169.14411404,
       166.96731984, 164.43341315, 153.41661552, 159.29756337,
       169.74084955, 152.79626732, 160.91145284, 162.28

In [160]:
def heights(i):
    if df['Sex'][i] == 'Female':
        height_female = rng.normal(165, 7)
        df['Height'][i] = height_female
    if df['Sex'][i] == 'Male':
        height_male = rng.normal(179, 8)
        df['Height'][i] = height_male

In [168]:
for i in range(100):
0    heights(i)

In [169]:
df

Age     Sex      Height Weight Alcohol_units_wk Smoker Cigs/day  \
0   NaN    Male  182.511263    NaN              NaN    NaN      NaN   
1   NaN    Male  161.462976    NaN              NaN    NaN      NaN   
2   NaN  Female  155.475587    NaN              NaN    NaN      NaN   
3   NaN  Female  155.475587    NaN              NaN    NaN      NaN   
4   NaN  Female  155.475587    NaN              NaN    NaN      NaN   
..  ...     ...         ...    ...              ...    ...      ...   
95  NaN  Female  155.475587    NaN              NaN    NaN      NaN   
96  NaN  Female  155.475587    NaN              NaN    NaN      NaN   
97  NaN  Female  155.475587    NaN              NaN    NaN      NaN   
98  NaN    Male  161.462976    NaN              NaN    NaN      NaN   
99  NaN    Male  161.462976    NaN              NaN    NaN      NaN   

   Time_Smoked_PrevExercise_wk Exercise_Consistency Diet  Calories_day  \
0                          NaN                  NaN  NaN          2249   
1                          NaN                  NaN  NaN             0   
2                          NaN                  NaN  NaN             0   
3                          NaN                  NaN  NaN             0   
4                          NaN                  NaN  NaN             0   
..                         ...                  ...  ...           ...   
95                         NaN                  NaN  NaN             0   
96                         NaN                  NaN  NaN             0   
97                         NaN                  NaN  NaN             0   
98                         NaN                  NaN  NaN             0   
99                         NaN                  NaN  NaN             0   

   Hours_Sleep  Water_l_day Stress_lvls Underlying_health  
0          NaN     2.186909         NaN               NaN  
1          NaN     0.824347         NaN               NaN  
2          NaN     1.037186         NaN               NaN  
3          NaN     1.664711         NaN               NaN  
4          NaN     1.533840         NaN               NaN  
..         ...          ...         ...               ...  
95         NaN     0.626702         NaN               NaN  
96         NaN     1.188805         NaN               NaN  
97         NaN     1.789016         NaN               NaN  
98         NaN     1.325402         NaN               NaN  
99         NaN     1.994363         NaN               NaN  

[100 rows x 15 columns]

In [261]:
df['Height'] = np.where(df['Sex']== 'Male', rng.normal(179, 8, size = sum(df['Sex']== 'Male')), rng.normal(165, 7, size = sum(df['Sex']== 'Female')))

ValueError: operands could not be broadcast together with shapes (100,) (44,) (56,) 

In [264]:
df['Height'] = np.where(df['Sex']== 'Male', rng.normal(179, 8), rng.normal(165, 7))

In [278]:
df['Height2'] = [rng.normal(179, 8) if x == 'Male' else rng.normal(165, 7) for x in df['Sex']]

In [279]:
df['Height2']

0     190.546329
1     179.750980
2     158.257570
3     166.622795
4     170.317099
         ...    
95    160.103413
96    155.273652
97    174.859319
98    175.684424
99    174.732721
Name: Height2, Length: 100, dtype: float64

In [280]:
df['Height']

0     168.941145
1     178.225199
2     160.479523
3     165.903232
4     155.679661
         ...    
95    164.364925
96    174.160846
97    166.722577
98    161.930362
99    158.182569
Name: Height, Length: 100, dtype: float64

List comprehension for below heights: https://stackoverflow.com/questions/19913659/pandas-conditional-creation-of-a-series-dataframe-column
    

In [296]:
def Heights():
    height = [rng.normal(179, 8) if x == 'Male' else rng.normal(165, 7) for x in df['Sex']]
    return height

ValueError: The truth value of a Series is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().

In [293]:
Heights()

Male


[182.76843356053558,
 179.1414593643049,
 152.8499068603903,
 153.2826569692336,
 156.6650759394108,
 167.28799078625318,
 175.15773373503663,
 157.0272677019508,
 165.68280312448894,
 154.22785660728408,
 156.67418866873652,
 171.0471227409297,
 155.47346499209632,
 167.33348036230237,
 168.39006233288913,
 185.6193964935473,
 161.6793216990794,
 188.26889474831205,
 167.97857976978534,
 170.18907866241057,
 188.71397212237832,
 170.09522845854823,
 159.89638438242778,
 171.91858704417453,
 170.428448422308,
 177.17539712603107,
 168.0689584292875,
 173.5541805927429,
 158.54186305660022,
 157.54377570556386,
 175.02553541767804,
 170.53660695775642,
 185.9314998051276,
 164.08871749727916,
 191.19100082762426,
 175.69227070189146,
 171.2497490752482,
 167.22801935902385,
 172.55590494433017,
 182.55222068850478,
 180.047840378947,
 160.8705949968801,
 175.83895299162296,
 182.6106067922192,
 169.68830929290831,
 179.49866981187486,
 173.43729845297415,
 173.2071428938761,
 177.776130

In [294]:
df['Sex']

0       Male
1       Male
2     Female
3     Female
4     Female
       ...  
95    Female
96    Female
97    Female
98      Male
99      Male
Name: Sex, Length: 100, dtype: object

In [132]:

heights(i)

C:\Users\James\Anaconda3\lib\site-packages\ipykernel_launcher.py:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  import sys


Average Heights: https://en.wikipedia.org/wiki/Average_human_height_by_country#Table_of_heights

ref: https://doi.org/10.1787%2F550623158455

In [20]:
height_male = rng.normal(179, 8, 100)
height_male

array([180.50066452, 178.1950648 , 187.17053501, 180.03026744,
       187.99744417, 167.87990078, 183.92444116, 193.79946907,
       178.47050409, 180.19132835, 181.18621648, 184.27880506,
       178.71694711, 187.22417559, 192.37420522, 179.02918474,
       184.71743354, 177.39077206, 183.5895585 , 188.64162811,
       187.09017613, 175.78427577, 186.63487036, 174.65100798,
       174.92279276, 194.423482  , 187.61329251, 168.19744211,
       181.87567186, 170.95915781, 178.27207958, 168.60114442,
       186.51368019, 174.7328704 , 181.52462932, 169.39411655,
       181.54852263, 187.17816928, 176.05386939, 170.84508983,
       180.84673919, 178.67493605, 183.87490525, 173.23047264,
       186.08216692, 185.19983505, 171.39496156, 184.07285691,
       172.87792912, 172.82447598, 178.94934027, 183.23668189,
       162.33483251, 159.39730723, 180.54523992, 191.61732754,
       174.47478266, 187.49116126, 178.23621383, 182.91118872,
       167.03243766, 174.96755763, 188.24371342, 178.99

Weight

I thought about using a predictor for the weight based off height,based on the below page and dataset comtained in it:
https://towardsdatascience.com/machine-learning-project-9-predict-weight-based-on-height-and-gender-9ed47285bcbbbut  

However I decide to make it a bit simpler and not over-complicate the project

In [24]:
drinker = ['NonDrinker', 'Drinker']
drinkers  = rng.choice(drinker,p=[0.02, 0.98], size = 100)

In [197]:
def drinker():
    drinker = ['NonDrinker', 'Drinker']
    drinkers  = rng.choice(drinker,p=[0.15, 0.85])
    return drinkers

In [216]:
df['Drinker'] = drinker()

In [217]:
df['Drinker'] 

0     Drinker
1     Drinker
2     Drinker
3     Drinker
4     Drinker
       ...   
95    Drinker
96    Drinker
97    Drinker
98    Drinker
99    Drinker
Name: Drinker, Length: 100, dtype: object

Average litres of pure alcohol drank in ireland per year: 11.6

Average litres of pure alcohol drank in ireland per week: 0.223 litres/week

https://www.oecd.org/ireland/Tackling-Harmful-Alcohol-Use-Ireland-en.pdf

In [38]:
litres_drank = rng.normal(0.223, 0.075 ,100)
units_drank= litres_drank /0.01
units_drank

array([12.70861569, 25.13055858, 21.17019232,  4.45538293, 35.80188195,
       24.91940299,  7.8948559 , 23.23271356, 17.31255494, 17.49539531,
       13.90314584, 28.76205927, 24.57050534, 25.21638462, 20.4105118 ,
       13.08354606, 29.42183085, 16.08952202, 17.14206939, 24.61434708,
       19.93602604, 24.1096712 , 16.12480796, 25.06126808, 34.22250617,
       36.11994029, 12.43887303, 22.88024548, 17.85241856, 34.17709766,
       10.15322667, 23.19961175, 32.35649598, 23.80547464, 26.16090964,
       23.26337308, 15.0884527 , 28.23292198,  9.02341786, 28.68197082,
       20.83947952, 11.31629824, 19.53249827, 26.94144318, 30.29515042,
       15.15468786, 26.86579268, 22.18346212, 30.69396233, 18.20845058,
       24.7314199 , 15.66020751, 36.78868058, 44.85911144, 27.62149419,
       16.29403461, 27.75230094, 26.65449223, 25.35346252, 20.86921481,
       28.45284715, 17.56273195, 25.58872514, 21.29369378, 30.17500277,
        9.36821196, 17.39834283, 22.01336013, 25.7895215 , 36.08

In [199]:
def units_drank():
    litres_drank = rng.normal(0.223, 0.075 ,100)
    unit_drank= litres_drank /0.01
    return unit_drank

In [ ]:
columns=['Age', 'Sex','Height', 'Weight', 'Drinker', 'Alcohol_units_wk','Smoker', 'Cigs/day', 'Exercise_wk', 'Diet',
                           'Calories_day','Hours_Sleep', 'Water_l_day']

In [312]:
def units_drank():
    units = [rng.normal(0.223, 0.075) if x == 'Drinker' else 0 for x in df['Drinker']]
    return units

In [319]:
len(units_drank()

100

In [240]:
while np.equal(df['Drinker'],'Drinker') == True
[rng.normal(179, 8) if x == 'Male' else rng.normal(165, 7) for x in df['Sex']]

SyntaxError: invalid syntax (<ipython-input-240-e2f444a96131>, line 1)

In [225]:
del df['drinker']
df

Age     Sex      Height Weight  Alcohol_units_wk Smoker Cigs/day  \
0   NaN    Male  182.511263    NaN         17.138568    NaN      NaN   
1   NaN    Male  161.462976    NaN         16.934330    NaN      NaN   
2   NaN  Female  155.475587    NaN         23.549604    NaN      NaN   
3   NaN  Female  155.475587    NaN         17.899985    NaN      NaN   
4   NaN  Female  155.475587    NaN         21.716816    NaN      NaN   
..  ...     ...         ...    ...               ...    ...      ...   
95  NaN  Female  155.475587    NaN         19.438220    NaN      NaN   
96  NaN  Female  155.475587    NaN         31.269708    NaN      NaN   
97  NaN  Female  155.475587    NaN         14.344123    NaN      NaN   
98  NaN    Male  161.462976    NaN         26.259684    NaN      NaN   
99  NaN    Male  161.462976    NaN         22.957349    NaN      NaN   

   Time_Smoked_PrevExercise_wk Exercise_Consistency Diet  Calories_day  \
0                          NaN                  NaN  NaN          2249   
1                          NaN                  NaN  NaN             0   
2                          NaN                  NaN  NaN             0   
3                          NaN                  NaN  NaN             0   
4                          NaN                  NaN  NaN             0   
..                         ...                  ...  ...           ...   
95                         NaN                  NaN  NaN             0   
96                         NaN                  NaN  NaN             0   
97                         NaN                  NaN  NaN             0   
98                         NaN                  NaN  NaN             0   
99                         NaN                  NaN  NaN             0   

   Hours_Sleep  Water_l_day Stress_lvls Underlying_health  Drinker  
0          NaN     2.186909         NaN               NaN  Drinker  
1          NaN     0.824347         NaN               NaN  Drinker  
2          NaN     1.037186         NaN               NaN  Drinker  
3          NaN     1.664711         NaN               NaN  Drinker  
4          NaN     1.533840         NaN               NaN  Drinker  
..         ...          ...         ...               ...      ...  
95         NaN     0.626702         NaN               NaN  Drinker  
96         NaN     1.188805         NaN               NaN  Drinker  
97         NaN     1.789016         NaN               NaN  Drinker  
98         NaN     1.325402         NaN               NaN  Drinker  
99         NaN     1.994363         NaN               NaN  Drinker  

[100 rows x 16 columns]

10g of pure alcohol = 1 unit, 10g = 0.01 litres

Smoker (cigarettes/day)

In [41]:
smoker = ['non_smoker', 'smoker']
smokers  = rng.choice(smoker,p=[0.83, 0.17], size = 100)

In [42]:
smokers

array(['non_smoker', 'non_smoker', 'non_smoker', 'non_smoker',
       'non_smoker', 'smoker', 'non_smoker', 'non_smoker', 'non_smoker',
       'non_smoker', 'non_smoker', 'non_smoker', 'smoker', 'smoker',
       'non_smoker', 'non_smoker', 'non_smoker', 'smoker', 'non_smoker',
       'smoker', 'non_smoker', 'smoker', 'non_smoker', 'non_smoker',
       'non_smoker', 'non_smoker', 'non_smoker', 'non_smoker',
       'non_smoker', 'smoker', 'non_smoker', 'non_smoker', 'non_smoker',
       'non_smoker', 'non_smoker', 'non_smoker', 'non_smoker',
       'non_smoker', 'non_smoker', 'smoker', 'non_smoker', 'non_smoker',
       'non_smoker', 'non_smoker', 'smoker', 'non_smoker', 'non_smoker',
       'non_smoker', 'smoker', 'non_smoker', 'non_smoker', 'non_smoker',
       'non_smoker', 'smoker', 'non_smoker', 'non_smoker', 'non_smoker',
       'smoker', 'smoker', 'non_smoker', 'non_smoker', 'non_smoker',
       'non_smoker', 'non_smoker', 'non_smoker', 'non_smoker', 'smoker',
       'non_smoker',

average number of cigarettes smoked: 12.5
    
https://www.hse.ie/eng/about/who/tobaccocontrol/tobaccofreeireland/adult-smoking-in-ireland.pdf - page 17

In [48]:
cigarettes_smoked = rng.normal(12.5, 5 ,100)
cigarettes_smoked

array([12.07392195, 16.6551315 , 13.59974079, 19.96986212,  7.51893133,
        5.6690298 , 11.18216767,  8.43551711, 14.09307207,  4.55222158,
       14.31116154, 18.67397442, 14.77103374, 15.00170152,  8.87332402,
       13.51737853, 12.76253879, 12.41731009,  1.92838953,  9.95438522,
       18.77167908,  9.7687632 ,  7.29094828, 14.56903088, 18.79956065,
        8.62939286, 10.46779685,  6.1306149 ,  7.11787624, 14.66191462,
       16.68755167, 16.36234506, 10.61077525, 20.53373885, 13.92466167,
       17.9687576 , 15.07943184, 17.01546881, 12.18246893, 26.43280287,
       16.04480688, 14.54488527, 12.58779798, 12.01065075, 17.01812546,
        7.04566542, 24.78964132,  7.28190092,  4.2435806 ,  6.5502805 ,
        8.0857096 , 11.02190547, 15.87868617,  8.10796501, 13.89136625,
        4.79666688, 15.0670103 , 12.37499517, 10.0469987 , 11.4030233 ,
       15.61390163, 10.68673816, 19.00453817, 17.30772429, 14.99952676,
       13.91557383, 14.48979185, 11.39452872,  9.7828431 , 16.03

Hours exercise per week

The National Guidelines for Physical Activity recommends that people should undertake at least 150 minutes of activity per week, made up of at least 30 minutes of moderate activity, five times a week.

https://www.hse.ie/eng/about/who/healthwellbeing/our-priority-programmes/heal/physical-activity-guidelines/

only 33% of Irish adults get sufficient excerise:

https://www.euro.who.int/__data/assets/pdf_file/0005/382514/ireland-eng.pdf

In [49]:
hours_exercised = rng.normal(1.5, 0.5 ,100)
hours_exercised

array([2.40048214, 1.11909559, 0.74049051, 1.38080929, 2.15200789,
       1.61543639, 1.0877618 , 1.52173311, 1.40477492, 1.95656003,
       1.39053999, 2.65205737, 1.42886358, 0.92067543, 1.66646254,
       1.00472095, 1.47539809, 0.74074158, 0.59233509, 1.27640871,
       1.22941225, 1.73790663, 1.10743119, 0.82285704, 1.24430371,
       0.66277117, 1.26137858, 0.50694725, 1.44050434, 1.15788232,
       1.75228711, 1.42048294, 1.4730339 , 0.64065844, 1.32309987,
       1.42925208, 0.82084647, 1.9651791 , 2.11592073, 1.81420057,
       0.35683892, 0.78046099, 0.9342632 , 1.30085831, 1.40132441,
       1.57719601, 1.48520116, 1.68682075, 1.10168398, 1.64681073,
       1.62061559, 1.20908862, 2.05765106, 1.45400102, 1.89564665,
       2.55273951, 1.61833835, 0.94357251, 1.17431069, 1.59224297,
       1.18645838, 1.51916491, 2.58569912, 1.07872061, 1.85996248,
       1.06272045, 0.9598799 , 1.50176493, 0.68499436, 1.69634875,
       1.21653017, 1.79117377, 0.77820811, 1.83822162, 1.39960

Diet

4 values -  vegetarian, fatty, sugary, balanced

In [52]:
diet_vals = ['Vegetarian', 'Fatty', 'Sugary', 'Balanced']
diet  = rng.choice(diet_vals,p=[0.10, 0.17, 0.33, 0.40], size = 100)
diet

array(['Vegetarian', 'Vegetarian', 'Balanced', 'Fatty', 'Balanced',
       'Balanced', 'Fatty', 'Sugary', 'Fatty', 'Vegetarian', 'Balanced',
       'Balanced', 'Sugary', 'Vegetarian', 'Sugary', 'Vegetarian',
       'Sugary', 'Sugary', 'Fatty', 'Balanced', 'Balanced', 'Sugary',
       'Fatty', 'Balanced', 'Balanced', 'Balanced', 'Balanced',
       'Balanced', 'Sugary', 'Fatty', 'Fatty', 'Sugary', 'Sugary',
       'Vegetarian', 'Vegetarian', 'Fatty', 'Sugary', 'Fatty', 'Fatty',
       'Sugary', 'Fatty', 'Balanced', 'Sugary', 'Sugary', 'Balanced',
       'Sugary', 'Fatty', 'Balanced', 'Balanced', 'Balanced', 'Balanced',
       'Vegetarian', 'Vegetarian', 'Sugary', 'Sugary', 'Fatty', 'Fatty',
       'Sugary', 'Sugary', 'Vegetarian', 'Sugary', 'Fatty', 'Balanced',
       'Balanced', 'Balanced', 'Fatty', 'Sugary', 'Balanced', 'Sugary',
       'Sugary', 'Balanced', 'Sugary', 'Sugary', 'Sugary', 'Sugary',
       'Sugary', 'Sugary', 'Sugary', 'Balanced', 'Balanced', 'Balanced',
       'Sugary',

Calories - we must adjust the calories each person intakes each day whether they are a male/female & their height.

Men - 2500
Women - 2000

https://www.nhs.uk/common-health-questions/food-and-diet/what-should-my-daily-intake-of-calories-be/

so for men over 180 cm we want to have them vary over 2500 calories, and women under 162 cm we will want them to vary under 2k, with some outliers in there


https://en.wikipedia.org/wiki/List_of_countries_by_food_energy_intake

In [314]:
for val, i in  df['Sex']:
    if val == 'Male':
        if df.loc[df['Height'][i] > 180:
            df['Calories_day'] = 2500 + rng.normal(100, 250)
        elif df['Height'] < 160:
            df['Calories_day'] = 2500 - rng.normal(200, 150)
        else:
            df['Calories_day'] = 2500 + rng.normal(100, 200)
    if val == 'Female':
        if df['Height'] > 170:
            df['Calories_day'] = 2000 + rng.normal(100, 250)
        elif df['Height'] < 155:
            df['Calories_day'] = 2000 - rng.normal(200, 150)
        else:
            df['Calories_day'] = 2000 + rng.normal(100, 200)

SyntaxError: invalid syntax (<ipython-input-314-34e0734539fa>, line 4)

In [317]:
def Calories():
    cals_in =  [2500+ rng.normal(75, 250) if x == 'Male' else 2000+ rng.normal(75, 250) for x in df['Sex']]
    return cals_in

In [318]:
Calories()

[2178.146503635701,
 2705.728503760054,
 2667.3002216363084,
 2720.4737889912753,
 2166.2839350076456,
 2594.898001516886,
 2692.2496649202594,
 2987.1815050406103,
 2434.1098639845845,
 2393.321393179668,
 2774.522506356,
 2288.2193034328334,
 2661.82972359391,
 2591.091678743286,
 2283.551558073557,
 2739.000867211839,
 2960.9803619932914,
 2349.640841560373,
 2805.534852831892,
 2915.008778449636,
 2201.776290313404,
 2417.030936446369,
 2472.396244650623,
 2569.416771092371,
 2811.508185198247,
 2613.137926967437,
 2826.483219623972,
 2932.2686504016274,
 2271.5807495319705,
 2724.765114527628,
 2330.5809795850555,
 2725.7164170818032,
 2594.359441665452,
 2715.2166890186068,
 2693.624997212111,
 2530.7356896415135,
 2602.0802737654644,
 2821.0022221439713,
 2310.171642665783,
 2613.9675085261515,
 2686.800125145767,
 2633.49405899266,
 2267.3535480536093,
 2620.3359585727408,
 3309.3782632126745,
 2586.722815449244,
 2896.423275487664,
 2533.4901800611738,
 2716.2628874466595,
 30

In [57]:
cals = df['Calories_day']
cals

Series([], Name: Calories_day, dtype: object)

Hours of sleep

7 to 9 hours of sleep is recommended for people with 705 of people meeting this requirement, 17% get too much and 14% get too little, this can accuratley be summised by a normal-(gaussian) dist with a mean of 8 and a SD of 1, however the results provided in the second wily link below actually provide us with the mean and SD in minutes, so it is an easy conversion to hours.

https://tilda.tcd.ie/news-events/2019/1910-sleep-older-adults/

https://agsjournals.onlinelibrary.wiley.com/doi/full/10.1111/jgs.16177


In [79]:
hours_slept = (rng.normal(463, 72.6, 100)/60)
hours_slept

array([ 8.96937233,  5.60533574,  7.53762386,  5.23891625,  7.30174565,
        6.31757381,  7.20101899,  7.41875487,  7.9681857 ,  7.9787149 ,
        7.55211698,  7.58707808,  6.71770174,  8.13811179,  7.55797381,
        5.89872559,  7.94240653,  7.45565345,  7.71449456,  8.29889394,
       10.64882742,  8.18246513,  6.44387234,  6.45400413,  7.19081657,
        6.3243152 ,  9.35209859,  9.23658697,  7.55924905,  9.16531541,
        8.46862953,  6.07499194,  8.80277985,  7.34366777,  6.97125183,
        7.42495714,  6.84978524,  8.19695644,  8.12232406,  7.97432874,
        9.4471117 ,  9.03193107,  7.40860833,  7.10233985,  8.07212949,
        6.57722912,  6.83150343,  6.36478716,  9.96773983,  6.5230222 ,
        8.78166807,  7.77159253,  5.18636381,  7.23954998,  7.2512868 ,
        7.88225325,  7.5200149 ,  6.60580894,  7.87211023,  6.92794042,
        9.01720028,  7.73246179,  9.29081975,  5.84402828,  7.24025218,
        7.37310076,  5.67573651,  6.64503237,  8.50661928,  8.30

Water Drank

We should consume 2 litres of water per day, however, from a quick google search I found that the Irish population do not consume this amount on average.

In [81]:
water_drank = rng.normal(1.5, 0.45, 100)

In [82]:
water_drank

array([2.18690915, 0.82434713, 1.03718579, 1.66471113, 1.53384042,
       1.54113053, 1.08607035, 1.80032959, 1.0336848 , 1.52643896,
       1.46882983, 1.67693685, 1.16182854, 1.75448403, 1.98082927,
       1.03516649, 0.89076178, 1.38513541, 1.36004696, 0.91317223,
       2.34307456, 0.78422224, 1.03694478, 1.19862763, 2.25429758,
       1.84883187, 0.91678057, 1.82178898, 2.19780387, 1.17360747,
       1.76087278, 2.27235699, 0.99203978, 1.1227781 , 1.62332423,
       1.65510109, 1.27002425, 1.26001003, 1.81004811, 0.99315831,
       0.47295625, 1.3436097 , 0.71040035, 1.04305022, 1.57422663,
       0.96177853, 1.62664453, 0.40601011, 1.38254204, 1.22190401,
       2.30359891, 1.29480584, 1.46306887, 1.15576607, 1.43461736,
       1.18783778, 1.22440639, 1.07085096, 1.50998113, 1.78817672,
       0.91665691, 1.74033912, 1.45144321, 1.31331315, 2.40151671,
       1.49904819, 0.44208341, 1.28171249, 2.42987943, 1.22087422,
       1.8251427 , 1.79791837, 1.67587167, 1.10150711, 1.18704

In [84]:
df['Water_l_day'] = water_drank

In [85]:
df['Water_l_day']

0     2.186909
1     0.824347
2     1.037186
3     1.664711
4     1.533840
        ...   
95    0.626702
96    1.188805
97    1.789016
98    1.325402
99    1.994363
Name: Water_l_day, Length: 100, dtype: float64

In [86]:
df['Sex'] = sex

In [ ]:
for val, i in  df['Sex']:
    if val == 'Male':
        if df.loc[df['Height'][i]] > 180:
            df.loc[df['Calories_day'][i]] = 2500 + rng.normal(100, 250)
        elif df['Height'] < 160:
            df['Calories_day'] = 2500 - rng.normal(200, 150)
        else:
            df['Calories_day'] = 2500 + rng.normal(100, 200)
    if val == 'Female':
        if df['Height'] > 170:
            df['Calories_day'] = 2000 + rng.normal(100, 250)
        elif df['Height'] < 155:
            df['Calories_day'] = 2000 - rng.normal(200, 150)
        else:
            df['Calories_day'] = 2000 + rng.normal(100, 200)

In [105]:
for val in  df['Sex']:
    if val == 'Male':
        if df['Height'][0] > 180:
            df.loc['Calories_day'][0] = 2500 + rng.normal(100, 250)

KeyError: 'Calories_day'

In [107]:
df.loc[0]

Age                                NaN
Sex                               Male
Height                         180.501
Weight                             NaN
Alcohol_units_wk                   NaN
Smoker                             NaN
Cigs/day                           NaN
Time_Smoked_PrevExercise_wk        NaN
Exercise_Consistency               NaN
Diet                               NaN
Calories_day                      2249
Hours_Sleep                        NaN
Water_l_day                    2.18691
Stress_lvls                        NaN
Underlying_health                  NaN
Name: 0, dtype: object

In [101]:
df['Calories_day']

0     0
1     0
2     0
3     0
4     0
     ..
95    0
96    0
97    0
98    0
99    0
Name: Calories_day, Length: 100, dtype: int64

In [95]:
df['Height'] = height_male

In [96]:
df['Height']

0     180.500665
1     178.195065
2     187.170535
3     180.030267
4     187.997444
         ...    
95    183.428961
96    183.885886
97    172.531322
98    178.478493
99    180.071472
Name: Height, Length: 100, dtype: float64